sycl/plugins/unified_runtime/ur/adapters/level_zero/kernel.cpp

```diff
@@ -618,9 +626,26 @@ UR_APIEXPORT ur_result_t UR_APICALL urKernelSetArgPointer(
                          ///< value is considered null.
 ) {
   std::ignore = Properties;
+  std::scoped_lock<ur_shared_mutex> Guard(Kernel->Mutex);
+
+  // OpenCL: "the arg_value pointer can be NULL or point to a NULL value
+  // in which case a NULL value will be used as the value for the argument
+  // declared as a pointer to global or constant memory in the kernel"
+  //
+  // We don't know the type of the argument but it seems that the only time
+  // SYCL RT would send a pointer to NULL in 'arg_value' is when the argument
+  // is a NULL pointer. Treat a pointer to NULL in 'arg_value' as a NULL.
+  void *PArgValue = const_cast<void *>(ArgValue);
+  if (ArgValue && *(void **)(const_cast<void *>(ArgValue)) == nullptr) {
+    PArgValue = nullptr;
+  }
```

Comment:
Is it really needed for USM pointers? It was needed for OpenCL buffers, true,, but why here?